sycl/include/CL/sycl/accessor.hpp

```diff
@@ -889,6 +889,7 @@ class accessor :
             getAdjustedMode(PropertyList),
             detail::getSyclObjImpl(BufferRef).get(), AdjustedDim, sizeof(DataT),
             BufferRef.OffsetInBytes, BufferRef.IsSubBuffer) {
+    checkDeviceAccessorBufferSize(BufferRef.get_count());
```

Comment:
Instead of calling `checkDeviceAccessorBufferSize` in each constructor, would it be better to call it once in the `AccessorBaseHost` constructor?

Reply:
For checking in AccessorBaseHost, we need to pass more information (e.g., Host Buffer) to AccessorBaseHost.